plugins/transformations/seq2array/srcCxx/Seq2ArrayTransformation.cxx

```diff
@@ -0,0 +1,481 @@
+            }
+        }
+        break;
+    }
+
+    case TypeKind::ARRAY_TYPE: {
+        if (output_type.kind() != TypeKind::ARRAY_TYPE) {
+            std::string error("Error: the ouput field <" + output_type.name()
+                    + "> is different to the input type <" + input_type.name());
+            throw std::runtime_error(error);
+        }
+
+        auto input_array_type = static_cast<const ArrayType &>(input_type);
+        auto output_array_type = static_cast<const ArrayType &>(output_type);
+
+        if (input_array_type.dimension_count() != output_array_type.dimension_count()) {
+            std::string error("Error: different dimension count of input and output "
+                    "array, input name: <" + input_array_type.name()
+                    + ">, output name <" + output_array_type.name() + ">: "
+                    + std::to_string(input_array_type.dimension_count())
+                    + " != " + std::to_string(output_array_type.dimension_count()));
+            throw std::runtime_error(error);
+        }
+
+        if (input_array_type.total_element_count() != output_array_type.total_element_count()) {
+            std::string error("Error: different total element count of input "
+                    "and output array, input name: <" + input_array_type.name()
+                    + ">, output name <" + output_array_type.name() + ">: "
+                    + std::to_string(input_array_type.dimension_count())
+                    + " != " + std::to_string(output_array_type.dimension_count()));
+            throw std::runtime_error(error);
+        }
+
+        if (input_array_type.content_type() == output_array_type.content_type()) {
+            return true;
+        } else {
+            // inspect element by element since it may be compatible
+            // (output with arrays and input with sequences)
+            switch (input_array_type.content_type().kind().underlying()) {
+            case TypeKind::STRUCTURE_TYPE:
+            case TypeKind::SEQUENCE_TYPE:
+            case TypeKind::ARRAY_TYPE:
+            case TypeKind::UNION_TYPE:
+                is_compatible = is_compatible && are_type_compatible(
+                        input_array_type.content_type(),
+                        output_array_type.content_type());
+                if (!is_compatible) {
+                    std::string error("Error: content of input member <"
+                            + input_array_type.name()
+                            + ">, not compatible with content of output member <"
+                            + output_array_type.name()
+                            + ">");
+                    throw std::runtime_error(error);
+                }
+                break;
+            // primitive type
+            default:
+                // not needed because if it is a primitive content_type, this
+                // has been already checked in the previous if condition
+                break;
```

Comment:
Wouldn't we get here also if the two members are primitive arrays of different types?

Reply:
no because if we have an input array of a primitive type, the output array of the primitive type must be the same, therefore, the previous `if` clause: `if (input_array_type.content_type() == output_array_type.content_type()) {` should have returned true

Reply:
good point, adding is_compatible = false